### PR TITLE
rds: Create a new high-level test for a particular scenario

### PR DESCRIPTION
**Pull request by draychev**
This PR adds a test, which confirms the bug discovered with Issue https://github.com/openservicemesh/osm/issues/1979.

A secondary goal for this PR is to start a new set of tests and tooling so that we can test what Envoy configuration would be generated based on a set of SMI policies (plus Services, Pods etc).

Here is the Envoy config for the bookstore proxy, which I believe points to the root cause for this:

image

```json
 {
     "version_info": "23",
     "route_config": {
      "@type": "type.googleapis.com/envoy.config.route.v3.RouteConfiguration",
      "name": "RDS_Outbound",
      "virtual_hosts": [
       {
        "name": "outbound_virtualHost|bookstore",
        "domains": [
         "bookstore.bookstore.svc",
         "bookstore.bookstore.svc.cluster",
         "bookstore.bookstore.svc.cluster.local",
         "bookstore.bookstore:80",
         "bookstore.bookstore.svc:80",
         "bookstore.bookstore.svc.cluster:80",
         "bookstore.bookstore.svc.cluster.local:80",
         "bookstore.bookstore"
        ],
        "routes": [
         {
          "match": {
           "headers": [
            {
             "name": ":method",
             "safe_regex_match": {
              "google_re2": {},
              "regex": ".*"
             }
            }
           ],
           "safe_regex": {
            "google_re2": {},
            "regex": ".*"
           }
          },
          "route": {
           "weighted_clusters": {
            "clusters": [
             {
              "name": "bookstore/bookstore",
              "weight": 100
             },
             {
              "name": "bookstore/bookstore-v1",
              "weight": 50
             },
             {
              "name": "bookstore/bookstore-v2",
              "weight": 50
             }
            ],
            "total_weight": 200
           }
          }
         }
        ]
       },
       {
        "name": "outbound_virtualHost|bookstore-v1",
        "domains": [
         "bookstore-v1.bookstore.svc.cluster",
         "bookstore-v1.bookstore.svc.cluster.local",
         "bookstore-v1.bookstore:80",
         "bookstore-v1.bookstore.svc:80",
         "bookstore-v1.bookstore.svc.cluster:80",
         "bookstore-v1.bookstore.svc.cluster.local:80",
         "bookstore-v1.bookstore",
         "bookstore-v1.bookstore.svc"
        ],
        "routes": [
         {
          "match": {
           "headers": [
            {
             "name": ":method",
             "safe_regex_match": {
              "google_re2": {},
              "regex": ".*"
             }
            }
           ],
           "safe_regex": {
            "google_re2": {},
            "regex": ".*"
           }
          },
          "route": {
           "weighted_clusters": {
            "clusters": [
             {
              "name": "bookstore/bookstore-v1",
              "weight": 50
             }
            ],
            "total_weight": 50
           }
          }
         }
        ]
       },
       {
        "name": "outbound_virtualHost|bookstore-v2",
        "domains": [
         "bookstore-v2.bookstore.svc:80",
         "bookstore-v2.bookstore.svc.cluster:80",
         "bookstore-v2.bookstore.svc.cluster.local:80",
         "bookstore-v2.bookstore",
         "bookstore-v2.bookstore.svc",
         "bookstore-v2.bookstore.svc.cluster",
         "bookstore-v2.bookstore.svc.cluster.local",
         "bookstore-v2.bookstore:80"
        ],
        "routes": [
         {
          "match": {
           "headers": [
            {
             "name": ":method",
             "safe_regex_match": {
              "google_re2": {},
              "regex": ".*"
             }
            }
           ],
           "safe_regex": {
            "google_re2": {},
            "regex": ".*"
           }
          },
          "route": {
           "weighted_clusters": {
            "clusters": [
             {
              "name": "bookstore/bookstore-v2",
              "weight": 50
             }
            ],
            "total_weight": 50
           }
          }
         }
        ]
       }
      ],
      "validate_clusters": true
     },
     "last_updated": "2020-11-03T22:44:54.512Z"
    }
   ]
  },
```

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

**Review by shashankram — APPROVED** [review]

File: tests/scenarios/helpers.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func makeService(kubeClient kubernetes.Interface, svcName string) {
+	// Create a service for the pod created above
```
Comment (snehachhabria) [inline tests/scenarios/helpers.go:22]:
```suggestion
	// Create a service for the pod created
```

File: tests/scenarios/helpers.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	selectors := map[string]string{
+		// These need to match the POD created above
```
Comment (snehachhabria) [inline tests/scenarios/helpers.go:24]:
```suggestion
		// These need to match the POD created 
```

**Review by snehachhabria — APPROVED** [review]

File: tests/scenarios/helpers.go (hunk 1 above)
Comment (draychev) [reply tests/scenarios/helpers.go:22]: https://github.com/openservicemesh/osm/pull/1993

File: tests/scenarios/helpers.go (hunk 2 above)
Comment (draychev) [reply tests/scenarios/helpers.go:24]: https://github.com/openservicemesh/osm/pull/1993

**Pull request merged**